### PR TITLE
AE-1425: Automate the linting, spell checking, and testing of PRs with GitHub Actions

### PR DESCRIPTION
**Pull request by Nomad-AE**
Added a couple of workflows for GitHub:

- eslint - Lints the files
- node.js - Runs `npm test` on each sub-directory
- typocispell - Is the spell checker that shows annotations on PRs
- vulnChecker - runs `npm audit` and npm outdated` and report vulnerable packages

Removed dependabot and also linted and upgraded some bots that were failing checks.

### CODE REVIEW

File: .github/.cspell.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    "autotasks",
+    "bignumber",
+    "calldatas",
+    "cbtc",
+    "ceth",
+    "codespell",
+    "collateralized",
+    "creds",
+    "disctribution",
+    "forta",
```
Comment (echoAE) [inline .github/.cspell.json:42]: Nit: `forta` is present twice.  Should one of them have a capital letter at the beginning? e.g. `Forta`

File: .github/.cspell.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "autotask",
+    "autotasks",
+    "bignumber",
+    "calldatas",
+    "cbtc",
+    "ceth",
+    "codespell",
+    "collateralized",
+    "creds",
+    "disctribution",
```
Comment (echoAE) [inline .github/.cspell.json:41]: Is this a typo from one of the packages that we use?

If so, can or should we disable spellcheck on any paths that have a `node_modules` directory?

Comment (Nomad-AE) [reply .github/.cspell.json:41]: yes, accidental typo accepted into the dictionary.

Comment (Nomad-AE) [reply .github/.cspell.json:41]: We are already ignoring the following: 
```  "ignorePaths": [
    "**/node_modules/**",
    "**/vscode-extension/**",
    "**/.git/**",
    ".vscode",
    "megalinter",
    "package-lock.json",
    "report]"```

File: .github/.cspell.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    "**/vscode-extension/**",
+    "**/.git/**",
+    ".vscode",
+    "megalinter",
+    "package-lock.json",
+    "report"
+  ],
+  "language": "en",
+  "noConfigSearch": true,
+  "words": [
```
[the remark this answers is not in the document]
Comment (Nomad-AE) [reply .github/.cspell.json:13]: yes it can, new words we lazily added to the bottom

File: asset-monitor/src/agent.spec.js [hunk 4]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
```
Comment (echoAE) [inline asset-monitor/src/agent.spec.js:3]: These ESLint checks should not be disabled. Each of the issues they pertain to seems valid.

I will make the changes and push.

**Review by echoAE — APPROVED** [review]
A few minor comments and nits. One quick question: why does the `eslint.yml` file specify checkout version 2.4.2, while all of the others use checkout version 3?

File: .github/.cspell.json (hunk 1 above)
Comment (Nomad-AE) [reply .github/.cspell.json:29]: sorted, lowedcased, and deduped in commit b12701395a494e5ca3ef3dd51c7c14423fe82fa1

File: .github/.cspell.json (hunk 2 above)
Comment (Nomad-AE) [reply .github/.cspell.json]: sorted, lowedcased, and deduped in commit b12701395a494e5ca3ef3dd51c7c14423fe82fa1

File: .github/.cspell.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nomad-AE) [reply .github/.cspell.json:13]: sorted, lowedcased, and deduped in commit b12701395a494e5ca3ef3dd51c7c14423fe82fa1

**Review by darian-ae — APPROVED** [review]

**Pull request merged**